### ISSUE TITLE
Prometheus: query_result() dashboard variables should go through the datasource backend

### ISSUE THREAD

**Issue by kostrse** [grafana/grafana#57137]
**What happened**:

Follow up on the #37784, #48790

Template variables in dashboards still query data via plugin proxy endpoint instead of using the Prometheus backend client.

![image](https://user-images.githubusercontent.com/1131253/196304854-a324310d-de6b-458d-bf10-f7aa419c7115.png)

As result of this, Azure authentication doesn't work for variables because it implemented in Prometheus client [here](https://github.com/grafana/grafana/blob/7f5914ff05074fdf9a5e8aacafa9703b9f4e6dfb/pkg/tsdb/prometheus/buffered/client.go#L41-L47) and since Grafana 9.1 was removed from plugin proxy (#50737).

```
Request URL: http://localhost:3001/api/datasources/proxy/59/api/v1/query
Request Method: POST
Status Code: 400 Bad Request

Authentication to data source failed
```

This appears to be specific to `query_result`.

**Environment**:
- Grafana version: 9.1 and main
- Data source type & version: Prometheus

**Comment by vishiy** [comment]
Is the impact only for query_result or other template functions as well ?

**Comment by taleena** [comment]
Per the linked issue above, it does look like template variables was part of this original list that might not have been implemented: https://github.com/grafana/grafana/issues/37784 (this implies it would affect template variables across the board).

**Comment by kostrse** [comment]
All other variables (documented [here](https://grafana.com/docs/grafana/latest/datasources/prometheus/#query-variable)) except `query_result(query)` work fine:

`label_names()` - works
`label_values(label)` - works
`metrics(metric)` - works
`query_result(query)` - **doesn't work**

Variables were implemented by  #49921, #50344

**Comment by kostrse** [comment]
All the mentioned above variables except `query_result(query)` call [metadataRequest()](https://github.com/grafana/grafana/blob/8fd4fcb987b88da08b9ec7f3ae7e72a3f97363ea/public/app/plugins/datasource/prometheus/datasource.tsx#L229) and this method was updated by #49921 to use endpoint URL `/api/datasources/${this.id}/resources${url}`. All of these variables work fine.

![image](https://user-images.githubusercontent.com/1131253/196563631-9502b76e-5b20-4be2-87f8-94dd9581388c.png)

`query_result(query)` calls different method [performInstantQuery](https://github.com/grafana/grafana/blob/8fd4fcb987b88da08b9ec7f3ae7e72a3f97363ea/public/app/plugins/datasource/prometheus/datasource.tsx#L645) and that method wasn't updated.

![image](https://user-images.githubusercontent.com/1131253/196563538-c8db619e-2aa6-43d8-b9b4-1e483b19d79a.png)

`performInstantQuery` uses default URL supplied by datasource instance settings from [frontendsettings.go](https://github.com/grafana/grafana/blob/10c080dad16c3e65e880a8000a9e290f37f2eb72/pkg/api/frontendsettings.go#L241):

![image](https://user-images.githubusercontent.com/1131253/196564170-b23830d9-c21c-4173-b292-f0129682860b.png)
